src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -1825,13 +1836,10 @@ public override BoundNode VisitConditionalOperator(BoundConditionalOperator node
                 //   object[] a = ...;
                 //   IEnumerable<object?> b = ...;
                 //   var c = true ? a : b;
+                BoundExpression consequencePlaceholder = CreatePlaceholderIfNecessary(consequence, consequenceResult);
+                BoundExpression alternativePlaceholder = CreatePlaceholderIfNecessary(alternative, alternativeResult);
                 HashSet<DiagnosticInfo> useSiteDiagnostics = null;
-                resultType = BestTypeInferrer.InferBestTypeForConditionalOperator(
-                    createPlaceholderIfNecessary(consequence, consequenceResult),
-                    createPlaceholderIfNecessary(alternative, alternativeResult),
-                    _conversions,
-                    out _,
-                    ref useSiteDiagnostics);
+                resultType = BestTypeInferrer.InferBestTypeForConditionalOperator(consequencePlaceholder, alternativePlaceholder, _conversions, out _, ref useSiteDiagnostics);
```

Comment:
>InferBestTypeForConditionalOperator [](start = 46, length = 35)

I expect that `isRef` would be passed into `BestTypeInferrer`. It feels like it should affect variance.